test/StarcounterLinqUnitTests/Tests/BaseTest.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using Starcounter;
+
+namespace StarcounterLinqUnitTests.Tests
+{
+    public class BaseTest
+    {
+        public void Run(Action action)
+        {
+            Exception exception = null;
+            Scheduling.ScheduleTask(() =>
```

Comment:
@un-tone it turns out that there is a method `Scheduling.RunTask` which does exactly what your wrapper does. It returns a standard .net `Task` and can be used as following:

`Scheduling.RunTask(() => { /* Do the job */ }).Wait()`.

**Note:** awaiting the task is not fully supported. In case of `await Scheduling.RunTask(() => {});` all the code after this line will be executed on a .net thread, but on on a Starcounter thread.

Reply:
It's good to know, thanks. Fixed it.